src/backend/columnar/cstore_tableam.c

```diff
@@ -376,22 +380,22 @@ cstore_fetch_row_version(Relation relation,
 						 Snapshot snapshot,
 						 TupleTableSlot *slot)
 {
-	elog(ERROR, "cstore_fetch_row_version not implemented");
+	elog(ERROR, "columnar_fetch_row_version not implemented");
```

Comment:
there is a distinction between `ereport` with `errcode(ERRCODE_FEATURE_NOT_SUPPORTED)` and `elog` in this file. Any reason for that? Or should we standardize on `ereport` with unsupported features throughout.

Reply:
ereport() is better for things users might encounter, and we want them to be as useful to the user as possible. elog() is for things we don't expect the user to reasonably encounter, or where we don't have much helpful information to include.

The most common kind of elog() that a user might actually see is a syscache lookup failure. Those normally shouldn't happen, but they still can, and normally don't have a lot of useful context because they happen at a low level.